frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx

```diff
@@ -0,0 +1,289 @@
+'use client';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { RootState } from '../../../..//store/store';
+import { parseBalance } from '@/utils/denom';
+import { getBalances } from '@/store/features/bank/bankSlice';
+import { getIBCBalances } from '@/utils/ibc';
+import chainDenoms from '@/utils/chainDenoms.json';
+import { useAppDispatch, useAppSelector } from '@/hooks/StateHooks';
+import {
+  getDelegations,
+  getAllValidators,
+} from '@/store/features/staking/stakeSlice';
+import { USD_CURRENCY } from '@/utils/contants';
+
+const chainDenomsData = chainDenoms as AssetData;
+
+const OverviewPage = () => {
+  const dispatch = useAppDispatch();
+  const networks = useAppSelector((state: RootState) => state.wallet.networks);
+  const stakingChains = useAppSelector(
+    (state: RootState) => state.staking.chains
+  );
+  const balanceChains = useAppSelector(
+    (state: RootState) => state.bank.balances
+  );
+  const nameToChainIDs: Record<string, string> = useAppSelector(
+    (state: RootState) => state.wallet.nameToChainIDs
+  );
+  const chainIDs = Object.keys(nameToChainIDs).map(
+    (chainName) => nameToChainIDs[chainName]
+  );
+
+  const chainIdToNames = useMemo(() => {
+    const chainIdToNames: Record<string, string> = {};
+    for (const key in nameToChainIDs) {
+      chainIdToNames[nameToChainIDs[key]] = key;
+    }
+    return chainIdToNames;
+  }, [nameToChainIDs]);
+
+  const convertToDollars = (denom: string, amount: number) => {
+    // todo: after common slice
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    const tokensPriceInfo: any = {};
+    if (tokensPriceInfo?.[denom]) {
+      const price = +tokensPriceInfo?.[denom]?.info?.[USD_CURRENCY] || 0;
+      return amount * price;
+    }
+    return 0;
+  };
+
+  const getDenomInfo = useCallback(
+    (chainID: string): { denom: string; decimals: number } => {
+      const currency = networks?.[chainID]?.network?.config?.currencies?.[0];
```

Comment:
```suggestion
      const currency = networks.[chainID].network.config.currencies.[0];
```

Ditto here. If chain information exists, then currency also exists, as we enforce this in the network details object.